addons/mail/models/mail_activity_type.py

```diff
@@ -125,3 +125,51 @@ def _inverse_triggered_next_type_id(self):
                 activity_type.chaining_type = 'trigger'
             else:
                 activity_type.chaining_type = 'suggest'
+
+    def write(self, values):
+        """ Protect some master types against deletiong or modification when they
+        are used as default in apps, in critical flows, plans, ... """
+        if values.get('res_model'):
+            xmlid_to_model = self._get_xmlid_model_link()
+            modified = self.browse()
+            for xml_id, model in xmlid_to_model.items():
+                activity_type = self.env.ref(xml_id, raise_if_not_found=False)
+                if activity_type and values['res_model'] != model:
+                    modified |= activity_type
+            modified = modified & self
```

Comment:
This is a strange way to proceed, as you're building a "modified" that is a recordset of all master data that would potentially be "wrongly modified" by this write, but they are not actually "modified" until you do an intersection.

Why not change the condition to `if activity_type and activity_type in self and values['res_model'] != model:` to make it a bit more simple?